editor/code_editor.cpp

```diff
@@ -1290,90 +1290,92 @@ void CodeTextEditor::convert_case(CaseStyle p_case) {
 void CodeTextEditor::move_lines_up() {
 	text_editor->begin_complex_operation();
 
-	Vector<int> carets_to_remove;
-
 	Vector<int> caret_edit_order = text_editor->get_caret_index_edit_order();
-	for (int i = 0; i < caret_edit_order.size(); i++) {
-		int c = caret_edit_order[i];
-		int cl = text_editor->get_caret_line(c);
 
-		bool swaped_caret = false;
-		for (int j = i + 1; j < caret_edit_order.size(); j++) {
-			if (text_editor->has_selection(caret_edit_order[j])) {
-				if (text_editor->get_selection_from_line() == cl) {
-					carets_to_remove.push_back(caret_edit_order[j]);
-					continue;
-				}
+	// Lists of carets representing each group
+	Vector<Vector<int>> caret_groups;
+	Vector<std::pair<int, int>> group_borders;
 
-				if (text_editor->get_selection_to_line() == cl) {
-					if (text_editor->has_selection(c)) {
-						if (text_editor->get_selection_to_line(c) != cl) {
-							text_editor->select(cl + 1, 0, text_editor->get_selection_to_line(c), text_editor->get_selection_to_column(c), c);
-							break;
-						}
-					}
+	// Search for groups of carets and their selections residing on the same lines
+	for (int i = 0; i < caret_edit_order.size(); i++) {
+		int c = caret_edit_order[i];
 
-					carets_to_remove.push_back(c);
-					i = j - 1;
-					swaped_caret = true;
-					break;
-				}
+		Vector<int> new_group{ c };
+		std::pair<int, int> group_border;
+		if (text_editor->has_selection(c))
+			group_border = std::make_pair<int, int>(text_editor->get_selection_from_line(c), text_editor->get_selection_to_line(c));
```

Comment:
Always use braces please.

Reply:
Sorry, I was pointed to the same mistake int a different PR but forgot to add them here